src/utils/helpers/application/getSectionsDetails.ts

```diff
@@ -1,22 +1,18 @@
-import {
-  ApplicationSection,
-  ApplicationSectionsConnection,
-  TemplateElement,
-  TemplateSection,
-  TemplateSectionsConnection,
-} from '../../generated/graphql'
+import { ApplicationSection, TemplateElement, TemplateSection } from '../../generated/graphql'
 import { SectionDetails } from '../../types'
 
-export const getTemplateSections = (sectionsConnection: TemplateSectionsConnection) => {
-  const sections = sectionsConnection?.nodes as TemplateSection[]
-  return getSectionDetailsArray(sections)
+export const getTemplateSections = (templateSections: TemplateSection[]) => {
+  // TODO: Sort template sections
+  // const sections = templateSections.sort(
+  //   ({ index: aIndex }, { index: bIndex }) => (aIndex as number) - (bIndex as number)
+  // )
+  return getSectionDetailsArray(templateSections)
 }
 
-export const getApplicationSections = (sectionsConnection: ApplicationSectionsConnection) => {
-  const applicationSections = sectionsConnection?.nodes as ApplicationSection[]
+export const getApplicationSections = (applicationSections: ApplicationSection[]) => {
   const sections = applicationSections
     .map(({ templateSection }: ApplicationSection) => templateSection as TemplateSection)
-    .sort((a, b) => (a.index as number) - (b.index as number))
+    .sort(({ index: aIndex }, { index: bIndex }) => (aIndex as number) - (bIndex as number))
```

Comment:
Well, I think that keeping this `index` is good, and that we should prevent indexes out of order in the Template builder - when we have that in place, just because it would be way simpler to see things in a proper sequence without gaps. And since this is not done by users it makes sense that the system can keep this up-to-date with the correct indexes.

Reply:
> we should prevent indexes out of order in the Template builder - when we have that in place, just because it would be way simpler to see things in a proper sequence without gaps.

Sure, in the Template Builder we can enforce this. But some users/developers might be manually editing exported templates, and you don't want to have to manually re-number every subsequent question/section if you delete an earlier one -- it's a massive hassle and very prone to mistakes (we've made several because of this when re-numbering Database IDs in GraphQL mutations)